routers/web/auth/oauth.go

```diff
@@ -1130,7 +1145,7 @@ func handleOAuth2SignIn(ctx *context.Context, source *auth.Source, u *user_model
 			SetLastLogin: true,
 		}
 		opts.IsAdmin, opts.IsRestricted = getUserAdminAndRestrictedFromGroupClaims(oauth2Source, &gothUser)
-		if err := user_service.UpdateUser(ctx, u, opts); err != nil {
+		if err := user_service.UpdateUser(ctx, user_model.NewAuthenticationSourceUser(), u, opts); err != nil {
```

Comment:
When a user login, the doer should be himself?

Reply:
Doesn't it look weird if the user changed its own admin and restricted status? Both are set by the authentication source.